From b2787f98146c20764ca528b181202f843d22f08b Mon Sep 17 00:00:00 2001
Message-Id: <b2787f98146c20764ca528b181202f843d22f08b.1555352526.git.liu.denton@gmail.com>
In-Reply-To: <cover.1555352526.git.liu.denton@gmail.com>
References: <3e3b9d6c8e54b326809b95adabd3be39d2d0770f.1555111963.git.liu.denton@gmail.com>
	<cover.1555352526.git.liu.denton@gmail.com>
From: Denton Liu <liu.denton@gmail.com>
Date: Mon, 15 Apr 2019 10:56:15 -0700
Subject: [PATCH v2 3/3] cocci: prevent extern function declarations
To: Git Mailing List <git@vger.kernel.org>
Cc: Junio C Hamano <gitster@pobox.com>,
    Jeff King <peff@peff.net>,
    Emily Shaffer <emilyshaffer@google.com>

Since function declarations already implicitly imply extern, create a
Coccinelle rule which removes the redundant extern.

Signed-off-by: Denton Liu <liu.denton@gmail.com>
---
 contrib/coccinelle/noextern.cocci | 6 ++++++
 1 file changed, 6 insertions(+)
 create mode 100644 contrib/coccinelle/noextern.cocci

diff --git a/contrib/coccinelle/noextern.cocci b/contrib/coccinelle/noextern.cocci
new file mode 100644
index 0000000000..8cb39ac947
--- /dev/null
+++ b/contrib/coccinelle/noextern.cocci
@@ -0,0 +1,6 @@
+@@
+type T;
+identifier f;
+@@
+- extern
+  T f(...);
-- 
2.21.0.921.gb27c68c4e9

